src/components/BatchClassComponents/DoubtClassExerciseComponent.js

```diff
@@ -0,0 +1,346 @@
+                <Typography
+                  align="center"
+                  mt={2}
+                  onClick={() => {
+                    setOpenDropOut(true);
+                  }}
+                  variant="body2"
+                  color="red"
+                  style={{ cursor: "pointer" }}
+                >
+                  can`t attend?
+                </Typography>{" "}
+              </>
+            ) : (
+              <Button
+                variant="contained"
+                fullWidth
+                style={{ marginTop: 20 }}
+                onClick={() => setConfirmationOpen(true)}
+              >
+                Enroll
+              </Button>
+            )}
+          </Box>
+        </Box>
+      </SwipeableDrawer>
+      <AlertDialog
+        open={ConfirmationOpen}
+        close={close}
+        title={actions?.title}
+        start_time={actions?.start_time}
+        end_time={actions?.end_time}
+        id={actions?.id}
+        exerciseReload={true}
+        setIsEnrolled={setIsEnrolled}
+        type="DoubtClass"
+      />
+    </div>
+  );
+};
+
+const DoubtClassExerciseComponent = (props) => {
+  const isActive = useMediaQuery("(max-width:" + breakpoints.values.sm + "px)");
+  const classes = useStyles();
+  const { actions, value } = props;
+  const [open, setOpen] = useState(false);
+  const [isEnrolled, setIsEnrolled] = useState(actions?.is_enrolled);
+  return !isEnrolled ? (
+    <>
+      <Box
+        backgroundColor="primary.light"
+        p={2}
+        mt={2}
+        sx={{ borderRadius: "20px" }}
+      >
+        <Typography
+          variant="body1"
+          mb={1}
+          align="left"
+          style={{
+            display: "flex",
+          }}
+        >
+          {" "}
+          <img
+            className={classes.icons}
+            src={require("./Revision/assets/Group.svg")}
+            alt="Students Img"
+          />
+          Need help? We got you covered. Enroll in the doubt class on{" "}
+          {format(actions?.start_time, "dd MMM yy")}
+          at {format(actions?.start_time, "hh:mm aaa")} -{" "}
+          {format(actions?.end_time, "hh:mm aaa")}
+        </Typography>
+        <div
+          style={{
+            width: "100%",
+            display: "flex",
+            justifyContent: "flex-end",
+            alignItems: "center",
+          }}
+        >
+          <Button
+            endIcon={<ArrowForwardIosIcon />}
+            onClick={() => {
+              setOpen(true);
+            }}
+            sx={{
+              width: isActive ? "90%" : "215px",
+              display: "flex",
+              justifyContent: "flex-end",
+            }}
+          >
+            View Class Details
+          </Button>
+        </div>
+      </Box>
+      <MoreDetails
+        open={open}
+        setOpen={setOpen}
+        actions={actions}
+        value={value}
+        isEnrolled={isEnrolled}
+        setIsEnrolled={setIsEnrolled}
+      />
```

Comment:
Can write this more compactly as:
```suggestion
      <MoreDetails {...{open, setOpen, actions, value, isEnrolled, setIsEnrolled}} />
```